### PR TITLE
Generate `nearbyint` for Round in tensorexpr llvm codegen, match `torch.round` result

### LINKED ISSUE #103465: `torch.round` behaves inconsistently when running a traced model
### 🐛 Describe the bug

I have a traced model which (given the same input) produces inconsistent output when compared to the native PyTorch code. This seems to happen on all versions of pytorch from 1.12.0 and up but not in 1.9.0. This issue disappears If I disable the fuser optimizer via `torch._C._jit_set_texpr_fuser_enabled(False)`.  I have also verified that the earliest version which is affected is 1.10.0 but did not try other versions in between 1.10.0 and 1.12.0.

After countless hours of debugging, it appears that the issue might be related to the `torch.round` function. So I came up with following minimal example
```python
import torch


def f1(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
    return ((x - y) / 2.0).round().int()


def f2(x: int, y: int) -> int:
    return int(round((x - y) / 2.0))


def main(enabled):
    print(f"Using torch version={torch.__version__} and optimization {enabled=}")
    example_inputs = (21, 16)
    for device in ["cpu", "cuda"]:
        out = f2(*example_inputs)

        example_inputs_tensor = tuple(torch.tensor(x).to(device) for x in example_inputs)
        out_torch = f1(*example_inputs_tensor)
        f1_traced = torch.jit.trace(f1, example_inputs=example_inputs_tensor)

        torch._C._jit_set_texpr_fuser_enabled(enabled)
        out_torch_traced = f1_traced(*example_inputs_tensor)

        print(f"[{device:4s}] Python = {out}, Torch = {out_torch}, Torch traced = {out_torch_traced}")


if __name__ == "__main__":
    for flag in True, False:
        main(enabled=flag)
```

Running this snippet with torch version 1.12.0, I can see that the output from the traced function is inconsistent with both PyTorch and native python. For some reason, it seems that, **when optimization is enabled, the traced function is rounding the result differently on CPU**.
```
Using torch version=1.12.0+cu113 and optimization enabled=True
[cpu ] Python = 2, Torch = 2, Torch traced = 3
[cuda] Python = 2, Torch = 2, Torch traced = 2
Using torch version=1.12.0+cu113 and optimization enabled=False
[cpu ] Python = 2, Torch = 2, Torch traced = 2
[cuda] Python = 2, Torch = 2, Torch traced = 2
```
while in version 1.9.0 all values are the same.
```
Using torch version=1.9.0+cu111 and optimization enabled=True
[cpu ] Python = 2, Torch = 2, Torch traced = 2
[cuda] Python = 2, Torch = 2, Torch traced = 2
Using torch version=1.9.0+cu111 and optimization enabled=False
[cpu ] Python = 2, Torch = 2, Torch traced = 2
[cuda] Python = 2, Torch = 2, Torch traced = 2
```

Is this a known issue? I am aware of #16498 and #17443. The documentation for `torch.round` mentions that the function [implements](https://pytorch.org/docs/stable/generated/torch.round.html) the "half-to-even" strategy, but this might be broken for traced functions?

### Versions
```
PyTorch version: 1.12.0+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.9.0 (default, Nov 15 2020, 14:28:56)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-5.14.0-1058-oem-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: 
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 2080 SUPER
GPU 1: NVIDIA GeForce RTX 2080 SUPER

Nvidia driver version: 510.108.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   43 bits physical, 48 bits virtual
CPU(s):                          32
On-line CPU(s) list:             0-31
Thread(s) per core:              2
Core(s) per socket:              16
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       AuthenticAMD
CPU family:                      23
Model:                           49
Model name:                      AMD Ryzen Threadripper PRO 3955WX 16-Cores
Stepping:                        0
Frequency boost:                 enabled
CPU MHz:                         2200.000
CPU max MHz:                     4402.7339
CPU min MHz:                     2200.0000
BogoMIPS:                        7785.03
Virtualization:                  AMD-V
L1d cache:                       512 KiB
L1i cache:                       512 KiB
L2 cache:                        8 MiB
L3 cache:                        64 MiB
NUMA node0 CPU(s):               0-31
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, STIBP conditional, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip rdpid overflow_recov succor smca sme sev sev_es

Versions of relevant libraries:
[pip3] memory-efficient-attention-pytorch==0.0.17
[pip3] mypy==1.3.0
[pip3] mypy-boto3-s3==1.26.127
[pip3] mypy-extensions==1.0.0
[pip3] mypy-protobuf==3.0.0
[pip3] numpy==1.23.4
[pip3] pytorch-ignite==0.2.0
[pip3] pytorch-lightning==1.3.8
[pip3] torch==1.12.0+cu113
[pip3] torch-scatter==2.0.9+av
[pip3] torch2trt==0.2.0
[pip3] torchmetrics==0.7.2
[pip3] torchvision==0.13.0+cu113
[conda] memory-efficient-attention-pytorch 0.0.17                   pypi_0    pypi
[conda] numpy                     1.23.4                   pypi_0    pypi
[conda] pytorch-ignite            0.2.0                    pypi_0    pypi
[conda] pytorch-lightning         1.3.8                    pypi_0    pypi
[conda] torch                     1.12.0+cu113             pypi_0    pypi
[conda] torch-scatter             2.0.9+av                 pypi_0    pypi
[conda] torch2trt                 0.2.0                    pypi_0    pypi
[conda] torchmetrics              0.7.2                    pypi_0    pypi
[conda] torchvision               0.13.0+cu113             pypi_0    pypi
```

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### PR DESCRIPTION
**Pull request by wenzhe-nrv**
Fixes #103465, which matches the behavior of `torch.round` ([doc](https://pytorch.org/docs/stable/generated/torch.round.html?highlight=round#torch.round)) - “round half to even” 

Using the repro code, the output is correct:
```
Using torch version=2.1.0a0+git84fedbc and optimization enabled=True
[cpu ] Python = 2, Torch = 2, Torch traced = 2
Using torch version=2.1.0a0+git84fedbc and optimization enabled=False
[cpu ] Python = 2, Torch = 2, Torch traced = 2
```

cc @EikanWang @jgong5

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]
LGTM. Can you add a UT to cover the fix per https://github.com/pytorch/pytorch/issues/103465?

File: torch/csrc/jit/tensorexpr/llvm_codegen.cpp
Comment (mirzadeh) [inline torch/csrc/jit/tensorexpr/llvm_codegen.cpp:1]: Should [line 2085](https://github.com/pytorch/pytorch/blob/ca337e434a2d8b7e342b7bb1f87ba6fe321dea9a/torch/csrc/jit/tensorexpr/llvm_codegen.cpp#L2085) also be modified to cover `ScalarType::Double`?

**Review by jgong5 — APPROVED** [review]

**Comment by wenzhe-nrv** [comment]
@malfet could you please help to check who should review and approve before merging? Thank you.

**Review by davidberard98 — APPROVED** [review]

**Comment by davidberard98** [comment]
thanks @wenzhe-nrv !

**Pull request closed without merging**
